lms/djangoapps/course_home_api/outline/v1/serializers.py

```diff
@@ -21,8 +22,35 @@ def get_url(self, tool):
         return request.build_absolute_uri(url)
 
 
+class CourseBlockSerializer(serializers.Serializer):
+    """
+    Serializer for Course Block Objects
+    """
+    blocks = serializers.SerializerMethodField()
+
+    def get_blocks(self, blocks):
+        return {
+            str(block_key): {
+                'id': str(block_key),
+                'type': block_key.category,
+                'display_name': blocks.get_xblock_field(block_key, 'display_name', block_key.category),
+                'lms_web_url': reverse(
+                    'jump_to',
+                    kwargs={'course_id': str(block_key.course_key), 'location': str(block_key)},
+                    request=self.context['request'],
+                ),
+                'children': [str(child_key) for child_key in blocks.get_children(block_key)],
+            }
+            for block_key in blocks
+        }
+
+
 class OutlineTabSerializer(serializers.Serializer):
     """
     Serializer for the Outline Tab
     """
     course_tools = CourseToolSerializer(many=True)
+    course_blocks = CourseBlockSerializer()
+    course_date_blocks = DateSummarySerializer(many=True)
+    dates_tab_link = serializers.CharField()
+    user_timezone = serializers.CharField()
```

Comment:
I want to see what the frontend looks like after the refactor, but we might end up writing a subserializer for the Dates Widget and that will return these three fields. I don't think I love some of these fields (dates_tab_link and user_timezone) being returned as first class fields.